libpod/container_exec.go

```diff
@@ -315,7 +315,17 @@ func (c *Container) ExecStartAndAttach(sessionID string, streams *define.AttachS
 		return err
 	}
 
-	c.newContainerEvent(events.Exec)
+	if c.HasHealthCheck() {
```

Comment:
I don't think this is safe, because it's still legal to do a `podman exec` into a container with a healthcheck, and we'd want a normal event there. You'll need to add an argument to `ExecStartAndAttach`. Personally, I'd suggest making the existing function `execStartAndAttach` with the extra argument, and adding an ExecStartAndAttach that calls execStartAndAttach with the argument always set so that a normal Exec event is generated